django/db/models/expressions.py

```diff
@@ -927,6 +927,89 @@ def convert_value(self, value, expression, connection, context):
         return value
 
 
+class SubQuery(Expression):
+    """
+    Notes about both SubQuery() and Exists().
+
+    MyModel.objects.annotate(
+        field=SubQuery(RelatedModel.objects.filter(date=Today())
+                                           .values('other_field')[:1]),
+        found=Exists(OtherModel.objects.filter(thing=True))
+    )
+
+    Note that for SubQuery, unless you are aggregating the sub query, you'll want to use
+    a [:1] on the subquery to use a LIMIT 1. The Exists() is not subject to this. You'll probably
+    also want to use a .values() with a single argument, unless you are using something that
+    will deal with multiple fields (like building a JSON object, maybe).
+
+    TODO: Multiple join fields.
+    TODO: Tests/example usage.
+    TODO: .filter(Exists()) support. Maybe .filter(field__where_exists=Exists()) ?
+    TODO: Auto-set output_field based on the subquery?
+
+    You can currently do it by .annotate(foo=Exists()).filter(foo=True)
+    """
+    template = '(%(subquery)s)'
+
+    def __init__(self, subquery, output_field=None, **extra):
+        self.subquery = subquery
+        self.extra = extra
+        super(SubQuery, self).__init__(output_field or self.output_field)
```

Comment:
What about inferring the `output_field` from the queryset?

Oh, that reminds me, what about when we don't return a field, but a record (with the intent of allowing something else to perform an operation on it)?